src/vs/workbench/contrib/terminalContrib/chatAgentTools/test/browser/bufferOutputPolling.test.ts

```diff
@@ -0,0 +1,117 @@
+	ensureNoDisposablesAreLeakedInTestSuite();
+
+	const defaultOriginalResult = { terminalExecutionIdleBeforeTimeout: false, output: '', pollDurationMs: PollingConsts.FirstPollingMaxDuration };
+	const defaultToken = CancellationToken.None;
+	const defaultLanguageModelsService = {} as any;
+	const defaultExecution = { getOutput: () => 'output' };
+
+	/**
+	 * Returns a set of arguments for racePollingOrPrompt, allowing overrides for testing.
+	 */
+	function getArgs(overrides?: {
+		pollFn?: () => Promise<ReturnType<typeof racePollingOrPrompt> extends Promise<infer R> ? R : any>;
+		promptFn?: () => { promise: Promise<boolean>; part?: Pick<ChatElicitationRequestPart, 'hide' | 'onDidRequestHide'> | undefined };
+		originalResult?: Parameters<typeof racePollingOrPrompt>[2];
+		token?: CancellationToken;
+		languageModelsService?: typeof defaultLanguageModelsService;
+		execution?: typeof defaultExecution;
+	}) {
+		return {
+			pollFn: overrides?.pollFn ?? (async () => ({ terminalExecutionIdleBeforeTimeout: true, output: 'output', pollDurationMs: 100 })),
+			promptFn: overrides?.promptFn ?? (() => ({ promise: new Promise<boolean>(() => { }), part: undefined })),
+			originalResult: overrides?.originalResult ?? defaultOriginalResult,
+			token: overrides?.token ?? defaultToken,
+			languageModelsService: overrides?.languageModelsService ?? defaultLanguageModelsService,
+			execution: overrides?.execution ?? defaultExecution
+		};
+	}
+
+	test('should resolve with poll result if polling finishes first', async () => {
+		let pollResolved = false;
+		const args = getArgs({
+			pollFn: async () => {
+				pollResolved = true;
+				return { terminalExecutionIdleBeforeTimeout: true, output: 'output', pollDurationMs: 100 };
+			}
+		});
+		const result = await racePollingOrPrompt(args.pollFn, args.promptFn, args.originalResult, args.token, args.languageModelsService, args.execution);
+		assert.ok(pollResolved);
+		assert.deepEqual(result, { terminalExecutionIdleBeforeTimeout: true, output: 'output', pollDurationMs: 100 });
+	});
+
+	test('should resolve with poll result if prompt is rejected', async () => {
+		const args = getArgs({
+			pollFn: async () => ({ terminalExecutionIdleBeforeTimeout: false, output: 'output', pollDurationMs: 100 }),
+			promptFn: () => ({ promise: Promise.resolve(false), part: undefined }),
+			originalResult: { terminalExecutionIdleBeforeTimeout: false, output: 'original', pollDurationMs: PollingConsts.FirstPollingMaxDuration }
+		});
+		const result = await racePollingOrPrompt(args.pollFn, args.promptFn, args.originalResult, args.token, args.languageModelsService, args.execution);
+		assert.deepEqual(result, args.originalResult);
+	});
+
+	test('should poll again if prompt is accepted', async () => {
+		let extraPollCount = 0;
+		const args = getArgs({
+			pollFn: async () => {
+				extraPollCount++;
+				return { terminalExecutionIdleBeforeTimeout: false, output: 'output', pollDurationMs: 100 };
```

Comment:
Is it possible to run this with faked timers like this?

https://github.com/microsoft/vscode/blob/0a7fbc80a6369de26e3ef003350cd75806da5bc4/src/vs/base/test/common/async.test.ts#L671

If not, we should rename this to `.integrationTest.ts` as unit tests are expected to execute very fast. The faster the test is the better as this will directly increase CI time, waiting for 100*20 for example isn't ideal.